### PR TITLE
Old solution for servers not suporting eval

### PR DESCRIPTION
**Pull request by inquam**
I see the parts of the code I once changed to fix the usage of eval on non eval supported servers don't look exactly the same. But I'd thought I'd still create this pull request and you could easily tailor it to the current code.

### CODE REVIEW

File: src/SerializableClosure.php [hunk 1]
```diff
@@ -174,11 +174,54 @@ private function reconstructClosure()
 
         // Evaluate the code to recreate the closure.
         if ($_fn = array_search(Serializer::RECURSION, $this->data['context'], true)) {
-            @eval("\${$_fn} = {$this->data['code']};");
+
+            // Eval work around
+            if(!$this->evalEnabled()) {
+                $secretKey = substr(str_shuffle("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"), 0, 1).substr(md5(time()), 1);
```
Comment (GrahamCampbell) [inline src/SerializableClosure.php]: this is not secure. Please use PHP 7's random generator.

**Comment by GrahamCampbell** [comment]
Maybe this code should be disabled by default, and opt-in via some kind of configuration?

**Comment by GrahamCampbell** [comment]
As in, if eval is not enabled, the default could be to thrown an exception, rather than to try this.

**Comment by jeremeamia** [comment]
The [stream wrapper technique](https://github.com/opis/closure/blob/master/src/ClosureStream.php) in [Opis Closure](https://github.com/opis/closure) is better. We should emulate that or just encourage people to use Opis Closure directly, considering that the API is mostly compatible.

**Comment by GrahamCampbell** [comment]
> The stream wrapper technique in Opis Closure

:+1:

**Pull request closed without merging**